FEWD/Week07/lesson13/classwork/responsive_articles/js/main.js

```diff
@@ -0,0 +1,26 @@
+if ($(window).width() >= 650) {
+  $("ul").show();
+  $("button").hide();
+} else {
+  $("ul").hide();
+  $("button").show();
+}
+
+function resize () {
+  var screenWidth = $(window).width();
+
+  if (screenWidth >= 650) {
+  $("ul").show();
+  $("button").hide();
+} else {
+  $("ul").hide();
+  $("button").show();
+}
+}
+
+function hamburgerMenu () {
+  $("ul").toggle();
```

Comment:
So as per the [documentation](http://api.jquery.com/toggle/):

"The matched elements will be revealed or hidden immediately, with no animation, by changing the CSS display property. If the element is initially displayed, it will be hidden; if hidden, it will be shown."

Like so:

![image](https://cloud.githubusercontent.com/assets/5670279/18264987/43f71764-740b-11e6-9fa1-4fdbbf21d39a.png)

But as we know, CSS classes are more flexible than modifying the `style` attribute of a HTML element, so try swapping this out for [toggleClass](http://api.jquery.com/toggleclass/) instead:

```js
function hamburgerMenu() {
  $("ul").toggleClass("visible");
}
```

... and then have a CSS class which does the display change:

```css
.visible {
  display: block;
}
```

(Don't forget to add the `display: none;` to the `<ul>`'s initial CSS so that the mobile nav is hidden to start.)

However by doing this, you've still got the mobile nav in bigger screen sizes. Again, much like before, rather than letting JS do the heavy lifting (with all your `$(window).width() >= 650` logic), let CSS media queries do what they were designed for:

```css
@media (min-width: 650px) {
  ul {
    display: block; /* to override the default display: hidden on the <ul> which hides it upon first page load on mobile */
  }

  .hamburger {
    display: none;
  }
}
```

And now that you've got it all working with CSS classes, you could try to add in some CSS animations (as per the jQuery documentation, a snippet of which I pasted above, you can't animate things you add directly onto the `style` attribute in the HTML).

Also by doing the showing/hiding of the `ul` and the hamburger `button` with CSS classes, you can get rid of everything in this JS file except the logic tied to handling the `click` and `toggleClass`. Much simpler.